GVFS/GVFS.UnitTests/Service/MacServiceTests.cs

```diff
@@ -0,0 +1,141 @@
+        private const string ActiveUserId = "502";
+        private static readonly string RegisteredRepo = Path.Combine("mock:", "code", "repo2");
+        private static readonly string ServiceDataLocation = Path.Combine("mock:", "registryDataFolder");
+
+        private MockFileSystem fileSystem;
+        private MockTracer tracer;
+        private Mock<MockPlatform> gvfsPlatformMock;
+
+        [SetUp]
+        public void SetUp()
+        {
+            this.tracer = new MockTracer();
+            this.fileSystem = new MockFileSystem(new MockDirectory(ServiceDataLocation, null, null));
+            this.gvfsPlatformMock = new Mock<MockPlatform>();
+            this.gvfsPlatformMock.Setup(p => p.GetCurrentUser()).Returns(ActiveUserId);
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            this.fileSystem = null;
+            this.tracer = null;
+        }
+
+        [TestCase]
+        public void ServiceStartTriggersAutoMountForCurrentUser()
+        {
+            Mock<RepoRegistry> repoRegistry = new Mock<RepoRegistry>();
+            repoRegistry.Setup(r => r.AutoMountRepos(It.IsAny<int>()));
+
+            GVFSService service = new GVFSService(
+                this.tracer,
+                GVFSServiceName,
+                startListening: false,
+                repoRegistry: repoRegistry.Object,
+                gvfsPlatform: this.gvfsPlatformMock.Object);
+
+            service.RunWithArgs(new string[] { $"--servicename={GVFSServiceName}" });
+
+            int expectedUserId = int.Parse(ActiveUserId);
+
+            repoRegistry.Verify(
+                r => r.AutoMountRepos(It.Is<int>(arg => arg == expectedUserId)),
+                Times.Once,
+                $"{nameof(repoRegistry.Object.AutoMountRepos)} was not called during Service startup");
+
+            repoRegistry.Verify(
+                r => r.TraceStatus(),
+                Times.Once,
+                $"{nameof(repoRegistry.Object.TraceStatus)} was not called during Service startup");
+        }
+
+        [TestCase]
+        public void RepoRegistryMountsOnlyRegisteredRepos()
+        {
+            Mock<GVFSMountProcess> mountProcessMock = new Mock<GVFSMountProcess>();
+            mountProcessMock.Setup(mp => mp.MountRepository(It.IsAny<string>(), It.IsAny<int>(), It.IsAny<ITracer>()));
+
+            this.CreateTestRepos(this.fileSystem, ServiceDataLocation);
+
+            MockPlatform gvfsPlatform = new MockPlatform();
+            RepoRegistry repoRegistry = new RepoRegistry(
+                this.tracer,
+                this.fileSystem,
+                ServiceDataLocation,
+                gvfsPlatform,
+                mountProcessMock.Object);
+
+            repoRegistry.AutoMountRepos(int.Parse(ActiveUserId));
+
+            mountProcessMock.Verify(
+                mp => mp.MountRepository(
+                    It.Is<string>(repo => repo.Equals(RegisteredRepo)),

```

Comment:
I can't figure out how the `repoRegistry` created on line 79 implies `RegisteredRepo` here. What bit of config have I missed?